ament_tools/build_types/cmake.py

```diff
@@ -324,15 +324,16 @@ def _get_command_prefix(self, name, context):
     def _get_command_prefix_windows(self, name, context):
         lines = []
         lines.append('@echo off\n')
-        lines.append('if defined AMENT_TRACE_SETUP_FILES echo Inside %~0')
         for path in context.build_dependencies:
             local_setup = os.path.join(path, 'local_setup.bat')
-            lines.append('if exist "{0}" call "{0}"\n'.format(local_setup))
+            lines.append(
+                'if "%AMENT_TRACE_SETUP_FILES%" NEQ "" echo call "{0}"'.format(local_setup))
+            lines.append('if exist "{0}" call "{0}"'.format(local_setup))
```

Comment:
None of the setup files does that. So no, it always echos what is supposed to be called and only calls it when it exists.